src/main/java/minijava/ir/optimize/JmpBlockRemover.java

```diff
@@ -26,33 +31,47 @@ public boolean optimize(Graph graph) {
 
   @Override
   public void visit(Block block) {
-    if (isJmpBlock(block) && hasOnePredecessor(block)) {
-      hasChanged = true;
-      remove(block);
-    }
+    tryToIdentifyNewEdge(block)
+        .ifPresent(
+            edge -> {
+              hasChanged = true;
+              remove(block, edge.v1, edge.v2, edge.v3);
+            });
   }
 
-  private boolean isJmpBlock(Block block) {
+  private Optional<Tuple3<Node, Integer, Block>> tryToIdentifyNewEdge(Block block) {
```

Comment:
It's just the combination of Optional and Tuple with the nested generics that makes this look pretty complicated. But just leave it. We have uglier things than this one in our code... ;)